variable/include/scipp/variable/transform.h

```diff
@@ -349,6 +349,15 @@ static void do_transform(Op op, Out &&out, Tuple &&processed, const Arg &arg,
       throw except::VariancesError("Variances in argument " +
                                    std::to_string(std::tuple_size_v<Tuple>) +
                                    " not supported.");
+    } else if constexpr (
+        std::is_base_of_v<
+            core::transform_flags::
+                expect_no_in_variance_if_out_cannot_have_variance_t,
+            Op> &&
+        !core::canHaveVariances<typename Out::value_type>()) {
```

Comment:
Can you add a test for this in `transform_test.cpp`? Other flags are tested there as well.